### PR TITLE
Fix accelerate failing tests

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do ?
This PR fixes some tests (~ 30) are that contained in the accelerate_tests flag. Recently, I ran the following workflow to have an overview of the failing tests : https://github.com/huggingface/transformers/actions/runs/9081904684 or the following CLI: `RUN_SLOW=True pytest -m accelerate_tests tests/ `
Related PR https://github.com/huggingface/transformers/pull/30808

Model fixed:
- BertModel
- ClipModel
- GPTSanJapaneseModel
- M2M100Model 
- MambaModel - same [fix](https://github.com/huggingface/transformers/pull/30418) as jamba 
- T5/MT5Model/UMT5Model - same fix since this is copied from code
- Whisper
- SiglipVisionModel for MP. As for the offload, I skipped the tests since fixing them would require to introduce a new attribute in PretrainedConfig (`_preload_module_classes` to be used in `dispatch_model`), not sure if it is worth it if this is the only model for now. 

Tests skipped:
- DbrxModel : because offload is not working and enabling it is not simple (core modeling code). However model parallelism works. 

See the run [here](https://github.com/huggingface/transformers/actions/runs/9116151177) with most of the tests passing  .  There are only 3 tests failing but I skipped them in the latest commit (siglipvisionmodel)
cc @ydshieh

### CODE REVIEW

File: src/transformers/models/bert/modeling_bert.py [hunk 1]
```diff
@@ -962,7 +962,7 @@ class BertModel(BertPreTrainedModel):
     `add_cross_attention` set to `True`; an `encoder_hidden_states` is then expected as an input to the forward pass.
     """
 
-    _no_split_modules = ["BertEmbeddings"]
+    _no_split_modules = ["BertEmbeddings", "BertLayer"]
```
Comment (ArthurZucker) [inline src/transformers/models/bert/modeling_bert.py:965]: Is this a general solution that will work for everyone? (I mean why was this failing the test)

**Review by ArthurZucker — APPROVED** [review]
Thanks. Approving but for the ssm cache I find it a bit weird that we have to do these x.to(). 
Also in general, accelerate is supposed to handle allllllll the devices no? 👿

File: src/transformers/models/bert/modeling_bert.py (hunk 1 above)
Comment (SunMarc) [reply src/transformers/models/bert/modeling_bert.py:965]: Yes that will work for everyone ! I don't know why the layer were not added to `_no_split_modules`, this is what we do for all models.

File: src/transformers/models/whisper/modeling_whisper.py [hunk 2]
```diff
@@ -1388,7 +1388,7 @@ def forward(
                 inputs_embeds, past_key_values_length=past_key_values_length, position_ids=position_ids
             )
 
-        hidden_states = inputs_embeds + positions
+        hidden_states = inputs_embeds + positions.to(inputs_embeds.device)
```
[the remark this answers is not in the document]
Comment (SunMarc) [reply src/transformers/models/whisper/modeling_whisper.py:1391]: `inputs_embeds` and `postions` can be on different devices (could be fixed if we find a way to make sure that they are on the same device, however, we would probably need to add another attribute in PretrainedConfig)

File: src/transformers/models/mamba/modeling_mamba.py [hunk 3]
```diff
@@ -244,6 +244,7 @@ def slow_forward(self, input_states, cache_params: Optional[MambaCache]=None):
         # 2. Convolution sequence transformation
         if cache_params is not None:
             ssm_state = cache_params.ssm_states[self.layer_idx].clone()
```
[the remark this answers is not in the document]
Comment (SunMarc) [reply src/transformers/models/mamba/modeling_mamba.py:246]: yes that's right !

**Review by ArthurZucker — APPROVED** [review]
Long due! Thanks for updating 😄

**Comment by SunMarc** [comment]
> Just wanna to know if those are failing for quite some time or it's a recent changes (in either transformers or accelerate) causing those failures? If it's the later case, could you elaborate it a bit more?

I'm not sure how long these tests have been failing. However, most of the tests were failing for a good reason (didn't put the right `no_split_modules`, modeling code not compatible)

**Review by ydshieh — APPROVED** [review]
ok, thanks~

**Pull request merged**